Library/Homebrew/brew.rb

```diff
@@ -116,8 +116,16 @@
     if Process.uid.zero? && !brew_uid.zero?
       tap_commands += %W[/usr/bin/sudo -u ##{brew_uid}]
     end
+    if help_flag
+      # Unset HOMEBREW_HELP to avoid confusing the tap
+      ENV.delete("HOMEBREW_HELP")
+    end
     tap_commands += %W[#{HOMEBREW_BREW_FILE} tap #{possible_tap}]
     safe_system(*tap_commands)
+    if help_flag
+      # Restore HOMEBREW_HELP after the tap
+      ENV["HOMEBREW_HELP"] = "1"
```

Comment:
`ENV["HOMEBREW_HELP"] = "1" if help_flag` and can skip the comment as it's clear without it, thanks.